### PR TITLE
feat(connector): [Authorizedotnet] Use connector_request_reference_id as reference to the connector

### PR DESCRIPTION
**Pull request by amitroy-thedev**
…ference to the connector

## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

Make any Payment for connector Authorizedotnet and see that you are getting "reference_id" field in the logs of payment request.

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by swangi-kumari** [comment]
Hello @amitroy-thedev ,

Pls address the CI fails

Run `cargo +nightly fmt` for formatting

**Comment by Rohxn16** [comment]
i wanna work on this issue

**Comment by swangi-kumari** [comment]
> i wanna work on this issue

Hey @Rohxn16 ,
 We appreciate your interest, but it seems that @amitroy-thedev  is already actively addressing this particular issue. However, we have several other open issues available for you to choose from. Feel free to let us know which issue among them you'd like to work on, and we'll be happy to assist you in any way we can.

**Comment by SamraatBansal** [comment]
Hey @amitroy-thedev Please tag the issue assigned to you with this PR

**Review by SamraatBansal — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/authorizedotnet/transformers.rs [hunk 1]
```diff
@@ -968,7 +971,9 @@ impl<F, Req>
                         mandate_reference: None,
                         connector_metadata: None,
                         network_txn_id: None,
-                        connector_response_reference_id: Some(transaction.transaction_id.clone()),
+                        connector_response_reference_id: Some(
+                                transaction_response.transaction_id.clone(),
+                            ),
                     }),
                     status: payment_status,
```
[the remark this answers is not in the document]
Comment (amitroy-thedev) [reply crates/router/src/connector/authorizedotnet/transformers.rs]: Hey @SamraatBansal 
i've made all the required changes please review it now
and so sorry for the delay and for the mistakes

File: crates/router/src/connector/authorizedotnet/transformers.rs [hunk 2]
```diff
@@ -323,6 +324,7 @@ impl TryFrom<&AuthorizedotnetRouterData<&types::PaymentsAuthorizeRouterData>>
             processing_options,
             subsequent_auth_information,
             authorization_indicator_type,
+            reference: item.router_data.connector_request_reference_id.clone(),
```
[the remark this answers is not in the document]
Comment (SamraatBansal) [reply crates/router/src/connector/authorizedotnet/transformers.rs]: Hey @amitroy-thedev I think you removed this change by mistake, as this is a required change, I asked to remove from `response` but `request` is required.
The only change here you would need to make is change the field name from `reference` to `ref_id`

**Comment by SamraatBansal** [comment]
Hey @amitroy-thedev Can you please address the unresolved comment

File: crates/router/src/connector/authorizedotnet/transformers.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (amitroy-thedev) [reply crates/router/src/connector/authorizedotnet/transformers.rs]: `ref_id: item.router_data.connector_request_reference_id.clone(),`
Do I replace with this code

[the remark this answers is not in the document]
Comment (amitroy-thedev) [reply crates/router/src/connector/authorizedotnet/transformers.rs]: @SamraatBansal 
and remove `pub reference: Option<String>` this code from response

**Comment by amitroy-thedev** [comment]
@SamraatBansal
Hello there,
Sir I've changed the code please review it and merge the code, hacktoberfest is going to be end 🥲

**Comment by amitroy-thedev** [comment]
@SanchithHegde 
Sir i've changed the required changes would you check it

**Comment by amitroy-thedev** [comment]
@SanchithHegde 
Sir i've changed the required changes would you please check it

**Comment by amitroy-thedev** [comment]
@SanchithHegde @Sangamesh26 @SamraatBansal 
This PR is being solved can you please review and merge it ..

**Review by deepanshu-iiitu — APPROVED** [review]

**Review by prasunna09 — APPROVED** [review]

**Review by SamraatBansal — APPROVED** [review]

**Comment by deepanshu-iiitu** [comment]
Hey @amitroy-thedev  ,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.  
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.  
Thanks!

**Pull request merged**